joern-cli/frontends/csharpsrc2cpg/src/main/scala/io/joern/csharpsrc2cpg/TypeMap.scala

```diff
@@ -114,38 +114,54 @@ class TypeMap(astGenResult: AstGenRunnerResult, initialMappings: List[NamespaceT
           case RecordDeclaration => true
           case _                 => false
       }
-      .map(parseClassDeclaration)
+      .map(classDecl => parseClassDeclaration(classDecl, namespaceName))
       .toSet
+
     namespaceName -> classes
   }
 
-  private def parseClassDeclaration(classDecl: DotNetNodeInfo): CSharpType = {
+  private def parseClassDeclaration(classDecl: DotNetNodeInfo, namespaceName: String): CSharpType = {
     val className = AstCreatorHelper.nameFromNode(classDecl)
+    val fullName  = s"${namespaceName}.${className}"
     val members = classDecl
       .json(ParserKeys.Members)
       .arr
       .map(AstCreatorHelper.createDotNetNodeInfo(_))
       .flatMap { x =>
         x.node match
-          case MethodDeclaration => parseMethodDeclaration(x)
-          case FieldDeclaration  => parseFieldDeclaration(x)
+          case MethodDeclaration => parseMethodDeclaration(x, fullName)
+          case FieldDeclaration  => parseFieldDeclaration(x, fullName)
           case _                 => List.empty
       }
       .toList
-    CSharpType(className, members.collectAll[CSharpMethod].l, members.collectAll[CSharpField].l)
+    CSharpType(fullName, members.collectAll[CSharpMethod].l, members.collectAll[CSharpField].l)
   }
 
-  private def parseMethodDeclaration(methodDecl: DotNetNodeInfo): List[CSharpMethod] = {
-    List(CSharpMethod(AstCreatorHelper.nameFromNode(methodDecl)))
+  private def parseMethodDeclaration(methodDecl: DotNetNodeInfo, fullName: String): List[CSharpMethod] = {
+    val params = methodDecl
+      .json(ParserKeys.ParameterList)
+      .obj(ParserKeys.Parameters)
+      .arr
+
+    val methodFullName = s"${fullName}.${AstCreatorHelper.nameFromNode(methodDecl)}"
+    val methodReturn   = AstCreatorHelper.createDotNetNodeInfo(methodDecl.json(ParserKeys.ReturnType)).code
+    val paramTypes = params
+      .map { param =>
+        param(ParserKeys.Type)(ParserKeys.Keyword)(ParserKeys.Value).str
+      }
+      .mkString(",")
+
+    val signatureParams = s"(${paramTypes})"
+    List(CSharpMethod(methodFullName, methodReturn, signatureParams))
```

Comment:
So only the type should have a full name, the type map compares only to the `name` of the method and fields when retrieved.